src/main.js

```diff
@@ -1,84 +1,73 @@
-// Private Methods
-function extendConfig(defaults, options) {
-    let prop, extended = {};
-
-    for (prop in defaults) {
-        if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
-            extended[prop] = defaults[prop];
-        }
-    }
-
-    for (prop in options) {
-        if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
-            extended[prop] = options[prop];
-        }
-    }
-
-    return extended;
-};
-
 export default class Concertina {
-    constructor(el, options) {
+    constructor(el, customOptions) {
         // Config
         this.element = el; // TODO: Get this to work with multiple elements - document.querySelectorAll('blah');
-        this.defaults = {
+        this.options = {
             'blockClass': '.Concertina',
             'tabClass': '.Concertina-tab',
             'tabActiveClass': '.is-selected',
             'panelClass': '.Concertina-panel',
             'panelActiveClass': '.is-visible'
         };
-        this.options = extendConfig.bind(this)(this.defaults, options);
-        this.tabs = this.element.querySelectorAll(this.options.tabClass);
-        this.panels = this.element.querySelectorAll(this.options.panelClass);
+
+        for (let prop in customOptions) {
+            this.options[prop] = customOptions[prop];
+        }
+
+        this.tabs = [...this.element.querySelectorAll(this.options.tabClass)];
+        this.panels = [...this.element.querySelectorAll(this.options.panelClass)];
     }
 
     init() {
         //TODO: Set aria-controls attr on tabs and aria-labelledby attr on panels.
+
         this.setTabsInactive();
         this.setPanelsInactive();
 
         this.element.addEventListener('click', (e) => {
-            if (e.target.classList.contains(this.options.tabClass.replace('.', ''))){
+            if (this.tabs.includes(e.target)){
                 e.preventDefault();
-
-                this.setTabsInactive();
-                this.setPanelsInactive();
                 this.setTabState(e.target);
-                this.setPanelState(e.target.getAttribute('href').replace('#', ''));
+                this.setPanelState(e.target.getAttribute('href'));
             }
         });
     }
 
-    // Disable all tabs.
-    setTabsInactive() {
-        for (let i = 0; i < this.tabs.length; i += 1) {
```

Comment:
I am one of those people. I prefer to err on the side of accomplishing an action with a single pattern, hence why I try stay away from the ++ operator. Not only does ++ have a different operator precedence to +=, despite accomplishing the exact same function, it's an extra thing that everybody has to think about. "When is it okay to use this? Should I use it? Maybe I'll use it elsewhere, since I'm just incrementing by one." I disagree with the benefits of reducing the line by four characters vs. the cognitive overhead of acceptable usage, which for juniors or people not familiar with JS is not going to be a consideration at the forefront of their mind.